civis/io/_tables.py

```diff
@@ -1029,8 +1029,16 @@ def civis_file_to_table(file_id, database, table, client=None,
 
     # Use Preprocess endpoint to get the table columns as needed
     # and perform necessary file cleaning
-    need_table_columns = ((not table_exists or existing_table_rows == 'drop')
+    table_columns_have_sql = True
+    if table_columns:
+        for col in table_columns:
+            if not col['sql_type']:
+                table_columns_have_sql = False
+                break
+
+    need_table_columns = (((not table_exists or existing_table_rows == 'drop')
```

Comment:
I'm now wondering why we shouldn't throw an error when the table_columns are provided but don't have a `sql_type`, seems to me we might want to throw an error rather than ignore the provided ones?

Reply:
@uttercm ahh do you mean because it'll override all of them, even though some of them might be provide?

Reply:
ya exactly, so they might be confused if the sql_types they did provide are different than the actual types we create since not all were provided

Reply:
That's a good point, but that would be a change in approach, so let me touch base with John on that.

Reply:
This makes sense. Just to be clear if no sql types are provided we will detect them but if some but not all are provided we will raise an error.

Reply:
Ah okay, I misunderstood the criteria in the first place. Thanks!